examples/shader/post_processing.rs

```diff
@@ -0,0 +1,256 @@
+) {
+    let window = windows.get_primary_mut().unwrap();
+    let size = Extent3d {
+        width: window.physical_width(),
+        height: window.physical_height(),
+        ..default()
+    };
+
+    // This is the texture that will be rendered to.
+    let mut image = Image {
+        texture_descriptor: TextureDescriptor {
+            label: None,
+            size,
+            dimension: TextureDimension::D2,
+            format: TextureFormat::Bgra8UnormSrgb,
+            mip_level_count: 1,
+            sample_count: 1,
+            usage: TextureUsages::TEXTURE_BINDING
+                | TextureUsages::COPY_DST
+                | TextureUsages::RENDER_ATTACHMENT,
+        },
+        ..default()
+    };
+
+    // fill image.data with zeroes
+    image.resize(size);
+
+    let image_handle = images.add(image);
+
+    let cube_handle = meshes.add(Mesh::from(shape::Cube { size: 4.0 }));
+    let cube_material_handle = materials.add(StandardMaterial {
+        base_color: Color::rgb(0.8, 0.7, 0.6),
+        reflectance: 0.02,
+        unlit: false,
+        ..default()
+    });
+
+    // The cube that will be rendered to the texture.
+    commands
+        .spawn_bundle(PbrBundle {
+            mesh: cube_handle,
+            material: cube_material_handle,
+            transform: Transform::from_translation(Vec3::new(0.0, 0.0, 1.0)),
+            ..default()
+        })
+        .insert(MainCube);
+
+    // Light
+    // NOTE: Currently lights are ignoring render layers - see https://github.com/bevyengine/bevy/issues/3462
+    commands.spawn_bundle(PointLightBundle {
+        transform: Transform::from_translation(Vec3::new(0.0, 0.0, 10.0)),
+        ..default()
+    });
+
+    // Main camera, first to render
+    commands.spawn_bundle(Camera3dBundle {
+        camera_3d: Camera3d {
+            clear_color: ClearColorConfig::Custom(Color::WHITE),
+        },
+        camera: Camera {
+            target: RenderTarget::Image(image_handle.clone()),
+            ..default()
+        },
+        transform: Transform::from_translation(Vec3::new(0.0, 0.0, 15.0))
+            .looking_at(Vec3::default(), Vec3::Y),
+        ..default()
+    });
+
+    // This specifies the layer used for the post processing camera, which will be attached to the post processing camera and 2d quad.
+    let post_processing_pass_layer = RenderLayers::layer((RenderLayers::TOTAL_LAYERS - 1) as u8);
+
+    let quad_handle = meshes.add(Mesh::from(shape::Quad::new(Vec2::new(
+        size.width as f32,
+        size.height as f32,
+    ))));
+
+    // This material has the texture that has been rendered.
+    let material_handle = post_processing_materials.add(PostProcessingMaterial {
+        source_image: image_handle,
+    });
+
+    // Post processing 2d quad, with material using the render texture done by the main camera, with a custom shader.
+    commands
+        .spawn_bundle(MaterialMesh2dBundle {
+            mesh: quad_handle.into(),
```

Comment:
Using a quad is fine but also suboptimal. A fullscreen triangle generated from vertex indices would be better, though drawing something which has no mesh entity is a bit more tricky I would think.

Reply:
I thought fullscreen triangle meant a triangle mesh covering all screen (so having corners outside of the screen). I guess I could modify the quad mesh into a triangle, compute its correct size and adapt uv values (or shader).

Your comment about “no mesh entity” makes me wonder if you meant something different ? A geometry shader ? Oh a compute shader applied to the render without needing a cpu mesh..? Sounds less trivial indeed

Reply:
By fullscreen triangle, I mean a triangle with one corner in one of the corners of the screen, one at the same y and twice the width, and one at the same x and twice the height. The uvs at those vertices are similarly 0,0, 2,0, and 0,2. This makes it so that the triangle just covers the screen and has correct uvs. But, these can be generated in the vertex shader from the vertex index alone so no vertex buffer is needed, just a draw command drawing indices 0..3 and then the vertex shader clip position is output as:
```
out.clip_position = vec4<f32>(f32(in_vertex_index & 1u), f32(in_vertex_index >> 1u), 0.5, 0.5) * 4.0 - 1.0;
```
Untested at this time of writing but that should do it.

Reply:
Can / should we add a explicit short hand for this? It seems like a common pattern.

Reply:
I suggested on Discord that we add a fullscreen pass node that can have its pipeline’s fragment shader stage and bind groups specialised, and that requires no mesh entity. I think for this PR just making a triangle mesh with index buffer containing 0,1,2, overriding the vertex stage with the above and calculating the uvs in the fragment stage should do it.

Reply:
Sounds good. We can get this in, and then watch it improve.

Reply:
Should I try this solution ? And try to benchmark the differences ? Maybe a different PR to make it clear we're not sure the added boilerplate is worth it (if we don't have shortcut available in engine).

Or would it be considered non blocking, add a documentation line about this concept, create an issue for future improvement ?

Reply:
My personal feeling is that this is nonblocking, and just needs an issue opened.

Reply:
I'm fine with that approach. If it were for internal bevy post-processing, I would block on it. But for an example, it's good enough.